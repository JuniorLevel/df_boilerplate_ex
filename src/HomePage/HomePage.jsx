// @flow

import React from 'react';
import { Layout } from '@/components';
import { sidebarButtons } from './data';

export const HomePage = (): React.Node => (
	<Layout title="Caption" buttons={sidebarButtons} theme="light" avatar={{ title: 'AV' }}>
		<div />
	</Layout>
);
