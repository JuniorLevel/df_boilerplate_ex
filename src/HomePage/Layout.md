```jsx { "file": "./HomePage.jsx" }

```

```jsx noeditor
import Image1 from './layout.png';
<img src={Image1} />;
```
