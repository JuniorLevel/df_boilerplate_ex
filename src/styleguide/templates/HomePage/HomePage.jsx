// @flow

import React from 'react';
import { Layout } from '@juniorlevel/df_boilerplate-library';

export const HomePage = (): React.Node => {
	const buttons = [
		{
			id: 1,
			title: 'Orders',
			onClick: () => {},
		},
		{
			id: 2,
			title: 'Compare',
			onClick: () => {},
		},
		{
			id: 3,
			title: 'Admin',
			onClick: () => {},
		},
	];

	return (
		<Layout
			title="Caption"
			buttons={buttons}
			theme="light"
			avatar={{ title: 'AV' }}
		>
			<div />
		</Layout>
	);
};
