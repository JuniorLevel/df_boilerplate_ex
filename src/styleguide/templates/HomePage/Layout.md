```jsx
import React from 'react';
import { Layout } from '@juniorlevel/df_boilerplate-library';

const buttons = [
	{
		id: 1,
		title: 'Orders',
		onClick: () => {},
	},
	{
		id: 2,
		title: 'Compare',
		onClick: () => {},
	},
	{
		id: 3,
		title: 'Admin',
		onClick: () => {},
	},
];

<Layout
	title="Caption"
	buttons={buttons}
	theme="light"
	avatar={{ title: 'AV' }}
>
	<div />
</Layout>;
```

```jsx noeditor
import Image1 from '../../assets/Layout/layout.png';
<img src={Image1} />;
```
