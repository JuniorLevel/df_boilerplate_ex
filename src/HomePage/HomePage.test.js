import React from 'react';
import { Context as ResponsiveContext } from 'react-responsive';
import { render } from '@testing-library/react';
import { HomePage } from './HomePage';
import '../../__mocks__/fileMock';

describe('HomePage', () => {
	test('should render HomePage component without crashing', () => {
		const { container } = render(<HomePage />);
		expect(container).toBeInTheDocument();
	});
	test('matches the snapshot desktop', () => {
		const { container: desktop } = render(
			<ResponsiveContext.Provider value={{ width: 1920 }}>
				<HomePage />
			</ResponsiveContext.Provider>
		);
		expect(desktop).toMatchSnapshot();
	});
	test('matches the snapshot laptop', () => {
		const { container: laptop } = render(
			<ResponsiveContext.Provider value={{ width: 1024 }}>
				<HomePage />
			</ResponsiveContext.Provider>
		);
		expect(laptop).toMatchSnapshot();
	});
	test('matches the snapshot mobile', () => {
		const { container: mobile } = render(
			<ResponsiveContext.Provider value={{ width: 360 }}>
				<HomePage />
			</ResponsiveContext.Provider>
		);
		expect(mobile).toMatchSnapshot();
	});
});
