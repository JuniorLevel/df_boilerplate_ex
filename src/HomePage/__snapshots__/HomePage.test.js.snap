// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage matches the snapshot desktop 1`] = `
<div>
  <div
    class="ant-layout Layoutstyles__StyledLayout-sc-httflb-0 ecYXYt css-5wsri9"
  >
    <div
      class="ant-layout css-5wsri9"
    >
      <main
        class="ant-layout-content Layoutstyles__StyledContent-sc-httflb-1 iIrHAO css-5wsri9"
      >
        <div />
      </main>
    </div>
  </div>
</div>
`;

exports[`HomePage matches the snapshot laptop 1`] = `
<div>
  <div
    class="ant-layout Layoutstyles__StyledLayout-sc-httflb-0 ecYXYt css-5wsri9"
  >
    <div
      class="ant-layout css-5wsri9"
    >
      <main
        class="ant-layout-content Layoutstyles__StyledContent-sc-httflb-1 iIrHAO css-5wsri9"
      >
        <div />
      </main>
    </div>
  </div>
</div>
`;

exports[`HomePage matches the snapshot mobile 1`] = `
<div>
  <div
    class="ant-layout Layoutstyles__StyledLayout-sc-httflb-0 ecYXYt css-5wsri9"
  >
    <div
      class="ant-layout css-5wsri9"
    >
      <main
        class="ant-layout-content Layoutstyles__StyledContent-sc-httflb-1 iIrHAO css-5wsri9"
      >
        <div />
      </main>
    </div>
  </div>
</div>
`;
